src/algebra/gcd_monoid/basic.lean

```diff
@@ -208,44 +211,53 @@ end associates
 supremum, `1` is bottom, and `0` is top. The type class focuses on `gcd` and we derive the
 corresponding `lcm` facts from `gcd`.
 -/
-@[protect_proj] class gcd_monoid (α : Type*) [comm_cancel_monoid_with_zero α] [nontrivial α]
-  extends normalization_monoid α :=
+@[protect_proj] class gcd_monoid (α : Type*) [comm_cancel_monoid_with_zero α] [nontrivial α] :=
 (gcd : α → α → α)
 (lcm : α → α → α)
 (gcd_dvd_left   : ∀a b, gcd a b ∣ a)
 (gcd_dvd_right  : ∀a b, gcd a b ∣ b)
 (dvd_gcd        : ∀{a b c}, a ∣ c → a ∣ b → a ∣ gcd c b)
-(normalize_gcd  : ∀a b, normalize (gcd a b) = gcd a b)
-(gcd_mul_lcm    : ∀a b, gcd a b * lcm a b = normalize (a * b))
+(gcd_mul_lcm    : ∀a b, associated (gcd a b * lcm a b) (a * b))
 (lcm_zero_left  : ∀a, lcm 0 a = 0)
 (lcm_zero_right : ∀a, lcm a 0 = 0)
 
+class normalized_gcd_monoid (α : Type*) [comm_cancel_monoid_with_zero α] [nontrivial α]
+  extends normalization_monoid α, gcd_monoid α :=
+(normalize_gcd  : ∀a b, normalize (gcd a b) = gcd a b)
+(normalize_lcm  : ∀a b, normalize (lcm a b) = lcm a b)
+
+
 export gcd_monoid (gcd lcm gcd_dvd_left gcd_dvd_right dvd_gcd  lcm_zero_left lcm_zero_right)
 
 attribute [simp] lcm_zero_left lcm_zero_right
 
 section gcd_monoid
-variables [comm_cancel_monoid_with_zero α] [nontrivial α] [gcd_monoid α]
+variables [comm_cancel_monoid_with_zero α] [nontrivial α]
 
-@[simp] theorem normalize_gcd : ∀a b:α, normalize (gcd a b) = gcd a b :=
-gcd_monoid.normalize_gcd
+@[simp] theorem normalize_gcd [normalized_gcd_monoid α] : ∀a b:α, normalize (gcd a b) = gcd a b :=
+normalized_gcd_monoid.normalize_gcd
 
-@[simp] theorem gcd_mul_lcm : ∀a b:α, gcd a b * lcm a b = normalize (a * b) :=
+theorem gcd_mul_lcm [gcd_monoid α] : ∀a b:α, associated (gcd a b * lcm a b) (a * b) :=
 gcd_monoid.gcd_mul_lcm
 
 section gcd
 
-theorem dvd_gcd_iff (a b c : α) : a ∣ gcd b c ↔ (a ∣ b ∧ a ∣ c) :=
+theorem dvd_gcd_iff [gcd_monoid α] (a b c : α) : a ∣ gcd b c ↔ (a ∣ b ∧ a ∣ c) :=
 iff.intro
   (assume h, ⟨h.trans (gcd_dvd_left _ _), h.trans (gcd_dvd_right _ _)⟩)
   (assume ⟨hab, hac⟩, dvd_gcd hab hac)
 
-theorem gcd_comm (a b : α) : gcd a b = gcd b a :=
+theorem gcd_comm [normalized_gcd_monoid α] (a b : α) : gcd a b = gcd b a :=
 dvd_antisymm_of_normalize_eq (normalize_gcd _ _) (normalize_gcd _ _)
   (dvd_gcd (gcd_dvd_right _ _) (gcd_dvd_left _ _))
   (dvd_gcd (gcd_dvd_right _ _) (gcd_dvd_left _ _))
 
-theorem gcd_assoc (m n k : α) : gcd (gcd m n) k = gcd m (gcd n k) :=
+theorem gcd_comm' [gcd_monoid α] (a b : α) : associated (gcd a b) (gcd b a) :=
```

Comment:
I would guess there's a similar lemma `gcd_assoc'`

Reply:
It's on L271, right?

Reply:
I added it after this comment, but forgot to resolve, sorry.